File: Hotelaria/src/main/java/com/dolts/hotelaria/dto/SearchRequestDTO.java
package com.dolts.hotelaria.dto;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

@Getter
@Setter
public class SearchRequestDTO implements Serializable {
    private String columnName;
    private String value;
    private Operation operation;
    private String joinTable;

    public enum Operation {
        EQUAL, LIKE, IN, GREATER_THAN, LESS_THAN, GREATER_THAN_EQUAL, LESS_THAN_EQUAL, BETWEEN, JOIN;
    }
}

File: hotelaria-frontend/src/app/shared/dto/searchRequestDTO.ts
export class SearchRequestDTO {
    columnName: string;
    value: string;
    operation: Operation;
    joinTable: string;
}

export enum Operation {
    EQUAL='EQUAL',
    LIKE='LIKE',
    IN='IN',
    GREATER_THAN='GREATER_THAN',
    LESS_THAN='LESS_THAN',
    GREATER_THAN_EQUAL='GREATER_THAN_EQUAL',
    LESS_THAN_EQUAL='LESS_THAN_EQUAL',
    BETWEEN='BETWEEN',
    JOIN='JOIN'
}
